File: backend-nestjs/auth-service/src/rbac/permission.entity.ts
import { Entity, Column, PrimaryColumn } from 'typeorm';

@Entity('permissions')
export class PermissionEntity {
  @PrimaryColumn({ type: 'bigint' })
  id: number;

  @Column({ length: 100 })
  name: string;

  @Column({ unique: true, length: 100 })
  code: string;

  @Column({ name: 'resource_id', type: 'bigint' })
  resourceId: number;

  @Column({ name: 'action_id', type: 'bigint' })
  actionId: number;
}

File: backend-java/cdm-auth/src/main/java/com/cdm/auth/entity/PermissionEntity.java
package com.cdm.auth.entity;

import com.baomidou.mybatisplus.annotation.IdType;
import com.baomidou.mybatisplus.annotation.TableId;
import com.baomidou.mybatisplus.annotation.TableName;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

@TableName("permissions")
@Getter @Setter @NoArgsConstructor
public class PermissionEntity {

    @TableId(type = IdType.ASSIGN_ID)
    private Long id;

    private String name;
    private String code;
    private Long resourceId;
    private Long actionId;
}
